### ISSUE TITLE
Use full pose dimension for neighbor future padding mask

### ISSUE THREAD

**Issue by Max-Bin** [tier4/Diffusion-Planner#189]
## Problem

The neighbor future padding mask is computed using only the first three columns:

```python
mask = torch.sum(torch.ne(neighbors_future[..., :3], 0), dim=-1) == 0
```

The same pattern exists in both training and validation paths. However, the model trajectory pose dimension is now `POSE_DIM = 4` (`x, y, cos(yaw), sin(yaw)`), and `heading_to_cos_sin()` is idempotent for already-4-column inputs.

For 4-column future tensors, the mask ignores the `sin(yaw)` column. This hardcodes the old 3-column `(x, y, yaw)` validity assumption into a 4-column trajectory pipeline.

## Impact

For legacy 3-column raw-yaw NPZs, this is mostly harmless because all available columns are checked before conversion.

For 4-column future tensors, the validity definition is incomplete. A timestep whose first three columns are zero but whose `sin(yaw)` column is non-zero would be incorrectly treated as padding. More generally, the code silently mixes old 3-column and new 4-column validity semantics.

The practical impact is lower than target-corrupting bugs, but this can still affect neighbor future loss, validation masking, and any metric that depends on `neighbors_future_valid`.

## Expected behavior

The padding mask should be computed consistently with the actual pose dimension:

```python
mask = torch.sum(torch.ne(neighbors_future, 0), dim=-1) == 0
```

or through a shared helper that handles both 3-column and 4-column inputs explicitly.

The same helper should be used by training, validation, and auxiliary evaluation scripts so future mask semantics do not drift between paths.
